app/api/units_api.rb

```diff
@@ -132,6 +134,14 @@ class UnitsApi < Grape::API
       end
     end
 
+    if unit_parameters[:portfolio_auto_generation_date]
```

Comment:
@ublefo can you move this into a validation of the unit that is triggered by any date change? e.g. change of start/end date or change of teaching period. Then this would trigger if you change any of those dates (though this may then cause challenges if you change the teaching period dates)

Reply:
https://guides.rubyonrails.org/active_record_validations.html

Reply:
See examples in unit and project

You can trigger this only when there are changes to a field:
```rb
class Project
    validate :tutorial_enrolment_same_campus, if: :will_save_change_to_enrolled?
end
```

And can also have custom if conditions
```rb
class Unit
   validates :code, uniqueness: { scope: :teaching_period, message: "%{value} already exists in this teaching period" }, if: :has_teaching_period?
end
```

Reply:
Yes, all the validation stuff from the old PR will be removed and converted to ActiveRecord validations. The "is date within teaching period" check can probably be ignored since we are already validating the start/end dates if a teaching period is defined.